astroquery/solarsystem/pds/core.py

```diff
@@ -0,0 +1,368 @@
+        epoch : `~astropy.time.Time` object, or str in format YYYY-MM-DD hh:mm, optional.
+                If str is provided then UTC is assumed.
+                If no epoch is provided, the current time is used.
+        location : array-like, or `~astropy.coordinates.EarthLocation`, optional
+            Observer's location as a
+            3-element array of Earth longitude, latitude, altitude, or
+            `~astropy.coordinates.EarthLocation`.  Longitude and
+            latitude should be anything that initializes an
+            `~astropy.coordinates.Angle` object, and altitude should
+            initialize an `~astropy.units.Quantity` object (with units
+            of length).  If ``None``, then the geocenter is used.
+        neptune_arcmodel : float, optional. which ephemeris to assume for Neptune's ring arcs
+            must be one of 1, 2, or 3 (see https://pds-rings.seti.org/tools/viewer3_nep.shtml for details)
+            has no effect if planet != 'Neptune'
+        get_query_payload : boolean, optional
+            When set to `True` the method returns the HTTP request parameters as
+            a dict, default: False
+
+
+        Returns
+        -------
+        response : `requests.Response`
+            The response of the HTTP request.
+
+
+        Examples
+        --------
+        >>> from astroquery.solarsystem.pds import RingNode
+        >>> import astropy.units as u
+        >>> systemtable, bodytable, ringtable = RingNode.ephemeris(planet='Uranus',
+        ...                 epoch='2024-05-08 22:39',
+        ...                 location = (-23.029 * u.deg, -67.755 * u.deg, 5000 * u.m))  # doctest: +REMOTE_DATA
+        >>> print(ringtable)  # doctest: +REMOTE_DATA
+              ring  pericenter ascending node
+                       deg          deg
+            ------- ---------- --------------
+                Six    293.129           52.0
+               Five    109.438           81.1
+               Four    242.882           66.9
+              Alpha    184.498          253.9
+               Beta     287.66          299.2
+                Eta        0.0            0.0
+              Gamma     50.224            0.0
+              Delta        0.0            0.0
+             Lambda        0.0            0.0
+            Epsilon    298.022            0.0
+        """
+
+        planet = planet.lower()
+        if planet not in ["mars", "jupiter", "saturn", "uranus", "neptune", "pluto", ]:
+            raise ValueError(
+                "illegal value for 'planet' parameter (must be 'Mars', 'Jupiter', 'Saturn', 'Uranus', 'Neptune', or 'Pluto')"
+            )
+
+        if isinstance(epoch, (int, float)):
+            epoch = Time(epoch, format='jd')
+        elif isinstance(epoch, str):
+            epoch = Time(epoch, format='iso')
+        elif epoch is None:
+            epoch = Time.now()
+            log.warning("Observation time not set. Using current time.")
```

Comment:
Using the current time if `epoch=None` is what the API says it does, so there is no need for a warning. But even if you do want to emit one you should use the Python warnings system instead of the logging system. The warnings from the logging system are meant for situations which the client application can do nothing about, but the function arguments are very much under the user's control. See also https://docs.python.org/3/howto/logging.html#when-to-use-logging

Reply:
I had implemented this using Python's warnings, but @bsipocz thought it was better as a logged warning... I guess I'm happy just deleting it altogether